internal/unionstore/art.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2024 TiKV Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//nolint:unused
+package unionstore
+
+import (
+	"math"
+	"sync"
+
+	tikverr "github.com/tikv/client-go/v2/error"
+	"github.com/tikv/client-go/v2/kv"
+)
+
+type ART struct {
+	sync.RWMutex
+	skipMutex       bool
+	allocator       artAllocator
+	root            artNode
```

Comment:
Perhaps wrapping the art related fields together, and it would be conveneient if we want to abstract it into a standalone library in the future.